.github/workflows/ci.yml

```diff
@@ -0,0 +1,86 @@
+name: CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest 
+
+    steps:
+      - uses: actions/checkout@v2
+      
+    #to speed up the execution time of Github Actions
+      - name: Cache multiple paths  
+        uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: |-
+            vendor
+            node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+                ${{ runner.os }}-yarn-
+     
+      - uses: getong/mariadb-action@v1.1  
+        with:
+          host port: 3306 
+          mariadb version: '10.5' 
+          mysql database: 'test' 
+      
+      - uses: ruby/setup-ruby@v1    
+        with:
+           ruby-version: 2.7.1
+           bundler-cache: true
+
+           
+      - name: Setup Node
+        uses: actions/setup-node@v1
+               
+      - run: 'export PATH=$PATH:`yarn global bin`'
+      
+      - run: export DB=test
+      
+      - run: export SENDER_EMAIL_ADDRESS='sender@wikiedu.org'
+      
+      - name: Installing Packages 
+        run: |
+            yarn install
+            gem install bundler
+            gem install rubocop
```

Comment:
I'm not sure that this and the following are necessary, since we call rubocop via `bundle exec`... so it should be installed as part of the bundle, and there's no need to also install it globally. At least, I think so.

Reply:
okay will try the test without these and see if this works or not but i remeber it was conflicting with the version of ruby and gems file that's why doing this made the dependencies work but i'll see to it

Reply:
this worked absolutely fine !!!!... Initially i was having trouble setting up correct ruby version that's why doing these bring the packages on even level but after installing correct ruby version , it installed the bundler and other gem files with same version from Gemfile.lock and also removing the above said things by you didn't make any difference in the workflow, thankx.